### PR TITLE
[usbdev,dv] usbdev stall_trans seq

### PR DESCRIPTION
**Pull request by Saad22386**
This PR includes the usbdev stall_tran_vseqs sequence, designed to validate the stall functionality of usbdev. Upon setting the out_stall bit for a OUT transaction, the DUT is expected to respond with PidTypeStall.

This builds on #22308. Merge that first, which should just leave the final commit from this PR.

### CODE REVIEW

**Comment by Saad22386** [comment]
@rswarbrick kindly review it.

File: hw/ip/usbdev/data/usbdev_testplan.hjson
Comment (rswarbrick) [inline hw/ip/usbdev/data/usbdev_testplan.hjson:664]: This is missing some quotes, which is causing CI failures. Other testpoints have what you need (e.g. `pkt_received`)

File: hw/ip/usbdev/dv/env/seq_lib/usbdev_stall_trans_vseq.sv
Comment (rswarbrick) [inline hw/ip/usbdev/dv/env/seq_lib/usbdev_stall_trans_vseq.sv:20]: Is this delay required? I don't think I can see why it would be.

Comment (Saad22386) [reply hw/ip/usbdev/dv/env/seq_lib/usbdev_stall_trans_vseq.sv]: If I remove this delay, it will also work. So, I am going to remove this delay.

**Review by rswarbrick — COMMENTED** [review]
This looks sensible to me. Looking at the testpoint, it talks about both `in_stall` and `out_stall`. Your sequence is just using `out_stall` and an output transaction. Don't we want to have an `in_stall`/`IN` mode as well?

**Comment by Saad22386** [comment]
> This looks sensible to me. Looking at the testpoint, it talks about both `in_stall` and `out_stall`. Your sequence is just using `out_stall` and an output transaction. Don't we want to have an `in_stall`/`IN` mode as well?

It could either be out_stall or in_stall. The functionality of in_stall has already been tested in another sequence (#21926) that has been merged.

**Comment by rswarbrick** [comment]
Ok. If I understand correctly, we're checking stalls for IN transactions in the `usbdev_in_stall` test, which implements the `in_stall` testpoint. Am I right in thinking that there is no point in the `stall_trans` mentioning OUT transactions? If so, I suggest you add an extra commit to this PR (probably before the existing one) which drops the generic nature of the testpoint.

**Comment by Saad22386** [comment]
> Ok. If I understand correctly, we're checking stalls for IN transactions in the `usbdev_in_stall` test, which implements the `in_stall` testpoint. Am I right in thinking that there is no point in the `stall_trans` mentioning OUT transactions? If so, I suggest you add an extra commit to this PR (probably before the existing one) which drops the generic nature of the testpoint.

ok I'm going to drop the generic nature of the test point.

**Comment by rswarbrick** [comment]
Thanks for the refactor to the testplan. You've also split it out into a separate PR (which seems sensible) and it is at #22308.

Reviewers: Merge that one first! :-)

**Review by rswarbrick — APPROVED** [review]
This looks sensible to me and I think this should be good to merge once #22308 has landed.

**Comment by Saad22386** [comment]
I've just rebased over some recent testplan tweaks that collide with it.

**Comment by Saad22386** [comment]
@rswarbrick can you merge this one?

**Pull request merged**
